### PR TITLE
[Doc] Quark quantization documentation

### LINKED ISSUE #12272: [Doc]: Add AMD Quark Documentation
### 📚 The doc issue

There is a lack of documentation about how to use [Quark](https://quark.docs.amd.com/latest/) to prepare models or what type of Quark quantized scheme that vLLM currently supports.

### Suggest a potential alternative/fix

Include a section about Quark or link to Quark documentation under `docs/source/features/quantization/fp8.md`

Extra thought:
Following the style in `docs/source/getting_started/installation/gpu` support tabs for different GPU backend quantization approach: CUDA, ROCm, XPU etc.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by cha557**
This PR supplement Quark quantization documentation.

FIX https://github.com/vllm-project/vllm/issues/12272

### CODE REVIEW

File: docs/source/features/quantization/quark.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+Quantization can effectively reduce memory and bandwidth usage, accelerate computation and improve
+throughput while with minimal accuracy loss. vLLM can leverage [Quark](https://quark.docs.amd.com/latest/),
+the flexible and powerful quantization toolkit, to produce performant quantized models to run on GPUs
+of AMD. Quark has specialized support for quantizing large language models with weight,
+activation and kv-cache quantization and along with the cutting-edge quantization algorithms like
+AWQ, GPTQ, Rotation and SmoothQuant.
+
+## Quark Installation
+
+Before quantizing model, you need to install Quark. Release of Quark can be installed with pip:
```
Comment (capnramses) [inline docs/source/features/quantization/quark.md:14]: Before quantizing 
-model
+models

File: docs/source/features/quantization/quark.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
+(quark)=
+
+# QUARK
```
Comment (capnramses) [inline docs/source/features/quantization/quark.md:3]: possibly "AMD Quark"

**Review by mgoin — APPROVED** [review]
LGTM, thank you!

File: docs/source/features/quantization/quark.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+Quantization can effectively reduce memory and bandwidth usage, accelerate computation and improve
+throughput while with minimal accuracy loss. vLLM can leverage [Quark](https://quark.docs.amd.com/latest/),
```
Comment (capnramses) [inline docs/source/features/quantization/quark.md:6]: -while
so it reads "throughput while with minimal accuracy loss"

File: docs/source/features/quantization/quark.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+the flexible and powerful quantization toolkit, to produce performant quantized models to run on GPUs
```
Comment (capnramses) [inline docs/source/features/quantization/quark.md:7]: suggest rephrase
"GPUs of AMD"
to
"AMD GPUs"

File: docs/source/features/quantization/quark.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+of AMD. Quark has specialized support for quantizing large language models with weight,
+activation and kv-cache quantization and along with the cutting-edge quantization algorithms like
```
Comment (capnramses) [inline docs/source/features/quantization/quark.md:9]: "and along with the" -> "and"

File: docs/source/features/quantization/quark.md (hunk 1 above)
Comment (capnramses) [inline docs/source/features/quantization/quark.md:14]: "Before quantizing a model" or "quantizing models".

"Release of Quark" -> "The latest release of Quark"

File: docs/source/features/quantization/quark.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+### 3. Set the Quantization Configuration
+
+We need to set the quantization configuration, you can check
+[quark config guide](https://quark.docs.amd.com/latest/pytorch/user_guide_config_description.html)
+for further details. Here we use FP8 per-tensor quantization on weight, activation,
+kv-cache and the quantization algorithm is autosmoothquant.
+
+:::{note}
+Note the quantization algorithm need json config file and the config file is located in
```
Comment (capnramses) [inline docs/source/features/quantization/quark.md:85]: "need json config file"
->
"needs a JSON config file"

File: docs/source/features/quantization/quark.md (hunk 2 above)
Comment (cha557) [reply docs/source/features/quantization/quark.md:3]: sure

File: docs/source/features/quantization/quark.md (hunk 3 above)
Comment (cha557) [reply docs/source/features/quantization/quark.md:6]: yes

**Review by capnramses — APPROVED** [review]
hope that helps - just made some grammar suggestions.
reads well!

i'm not 100% sure it matters, but it might read better with "llama" as "Llama" and autosmoothquant as "AutoSmoothQuant"

**Comment by cha557** [comment]
@capnramses all grammar issues fixed, thanks

File: docs/source/features/quantization/quark.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+)
+model.eval()
+
+tokenizer = AutoTokenizer.from_pretrained(MODEL_ID, model_max_length=MAX_SEQ_LEN)
+tokenizer.pad_token = tokenizer.eos_token
+```
+
+### 2. Prepare the Calibration Dataloader
+
+Quark use the [PyTorch Dataloader](https://pytorch.org/tutorials/beginner/basics/data_tutorial.html)
```
Comment (DarkLight1337) [inline docs/source/features/quantization/quark.md:55]:
```suggestion
Quark uses the [PyTorch Dataloader](https://pytorch.org/tutorials/beginner/basics/data_tutorial.html)
```
